pydantic/tools.py

```diff
@@ -77,3 +78,18 @@ def parse_raw_as(
         json_loads=json_loads,
     )
     return parse_obj_as(type_, obj, type_name=type_name)
+
+
+def create_partial_model(model: Type[BaseModel]) -> Type[BaseModel]:
+    """
+    Create a version of the model that is suitable for partial validation.
+    Useful when working with CRUD operations, and you need to use an
+    existing model for a PATCH operation.
+    """
+
+    class PartialModel(model):  # type: ignore
+        class Config(model.Config):  # type: ignore
+            partial = True
+
+    PartialModel.__name__ = model.__name__
```

Comment:
I'm curious, why do you make the model name the same as the original one ?

I personally think it's a bit confusing but I don't understand why you want to do it in the first place, so I might be missing a piece here

Reply:
@christophelec it would show up as `PartialModel` otherwise 🙂

Reply:
How about something like

```suggestion
    PartialModel.__name__ = f'Partial{model.__name__}'
```
?